pkg/util/clusterdata/clusterdata.go

```diff
@@ -164,3 +184,61 @@ type roundTripperFunc func(*http.Request) (*http.Response, error)
 func (r roundTripperFunc) RoundTrip(req *http.Request) (*http.Response, error) {
 	return r(req)
 }
+
+type cachingEnricher struct {
+	now      func() time.Time
+	log      *logrus.Entry
+	db       *database.Database
+	m        metrics.Interface
+	inner    OpenShiftClusterEnricher
+	cacheTTL time.Duration
+}
+
+func (e *cachingEnricher) Enrich(ctx context.Context, docs ...*api.OpenShiftClusterDocument) {
+	e.enrich(ctx, docs...)
+}
+
+func (e *cachingEnricher) enrich(ctx context.Context, docs ...*api.OpenShiftClusterDocument) []*api.OpenShiftClusterDocument {
+	now := e.now().UTC()
+	filteredDocs := make([]*api.OpenShiftClusterDocument, 0, len(docs))
+
+	for _, doc := range docs {
+		if doc.LastEnrichment == nil || now.Sub(*doc.LastEnrichment) > e.cacheTTL {
+			filteredDocs = append(filteredDocs, doc)
+		}
+	}
+
+	missCount := len(filteredDocs)
+	hitCount := len(docs) - len(filteredDocs)
+
+	e.m.EmitGauge("enricher.cache.hit.count", int64(hitCount), nil)
+	e.m.EmitGauge("enricher.cache.miss.count", int64(missCount), nil)
+
+	if missCount == 0 {
+		return nil
+	}
+
+	e.inner.Enrich(ctx, filteredDocs...)
+
+	for _, doc := range filteredDocs {
+		doc.LastEnrichment = &now
+	}
+
+	return filteredDocs
+}
+
+func (e *cachingEnricher) EnrichAndPersist(ctx context.Context, docs ...*api.OpenShiftClusterDocument) {
+	updatedDocs := e.enrich(ctx, docs...)
+
+	// Ideally we should perform bulk update here, but there is no easy and reliable way to do it.
+	// .NET and Java Cosmos DB libraries include support of bulk operations [1],
+	// but the implementation is very complex and it seems to rely on undocumented system stored procedures.
+	// [1] https://docs.microsoft.com/en-us/azure/cosmos-db/bulk-executor-overview
```

Comment:
+1.  Remember that we have HTTP keepalive/connection pooling, plus cosmos has a low latency SLA, and we are in small batches, so this is not by any means critical.